packages/flutter_tools/lib/src/cache.dart

```diff
@@ -76,6 +76,17 @@ class Cache {
     _lock = null;
   }
 
+  static String _dartSdkVersion;
+
+  static String get dartSdkVersion {
+    if (_dartSdkVersion == null) {
+      File versionFile = new File(path.join(flutterRoot, 'bin', 'cache', 'dart-sdk.version'));
```

Comment:
Maybe https://api.dartlang.org/stable/1.18.1/dart-io/Platform/version.html would be better?